OpenKeychain/src/main/java/org/sufficientlysecure/keychain/remote/OpenPgpService.java

```diff
@@ -869,6 +871,31 @@ private Intent checkRequirements(Intent data) {
             return result;
         }
 
+        // check if database has to be updated
+        result = databaseIsUpToDateOrReturnIntent(data);
+        if (result != null) {
+            return result;
+        }
+
+        return null;
+    }
+
+
+    private Intent databaseIsUpToDateOrReturnIntent(Intent data) {
```

Comment:
I like this idea! Always wondered how to do this nicely, but returning a pendingintent which somehow leads the user to the consolidation is a good idea